File: Problems/No.1984_Minimum_Difference_Between_Highest_and_Lowest_of_K_Scores/minimum_difference_between_highest_and_lowest_of_k_scores.py
class Solution(object):
    def minimumDifference(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """

        # parameter
        min_gap = float('inf')

        # start computing
        nums.sort()
        for i in range(len(nums)-k+1):
            min_gap = min(min_gap, nums[i+k-1]-nums[i])
        
        return min_gap

File: Problems/No.1984_Minimum_Difference_Between_Highest_and_Lowest_of_K_Scores/minimum_difference_between_highest_and_lowest_of_k_scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        
        // parameter
        int min_gap = INT_MAX;

        // start computing
        std::sort(nums.begin(), nums.end());
        for (int i=0; i<nums.size()-k+1; i++) 
            min_gap = std::min(min_gap, nums[i+k-1]-nums[i]);
        
        return min_gap;

    }
};
